source/image-handler/test/image-request/parse-image-bucket.spec.ts

```diff
@@ -126,4 +126,43 @@ describe("parseImageBucket", () => {
       });
     }
   });
+
+  it("should parse bucket-name from first part in thumbor request but fail since it's not allowed", () => {
+    // Arrange
+    const event = { path: "/filters:grayscale()/s3:test-bucket/test-image-001.jpg" };
+    process.env.SOURCE_BUCKETS = "allowedBucket001, allowedBucket002";
+
+    // Act
+    const imageRequest = new ImageRequest(s3Client, secretProvider);
+
+    const bucket = imageRequest.parseImageBucket(event, RequestTypes.THUMBOR);
+    // Assert
+    expect(bucket).toEqual("allowedBucket001")
+  })
+
+  it("should parse bucket-name from first part in thumbor request and return it", () => {
```

Comment:
It might be worthwhile adding a test to cover the newly fixed thumbor chaining issue: https://github.com/aws-solutions/serverless-image-handler/issues/343

It should have both the legacy and new chaining method

```
it("should parse bucket-name from first part in thumbor request and return it when using legacy multiple filters", () => {
    // Arrange
    const event = { path: "/filters:grayscale()/filters:rotate(180)/s3:test-bucket/test-image-001.jpg" };
    process.env.SOURCE_BUCKETS = "allowedBucket001, test-bucket";

    // Act
    const imageRequest = new ImageRequest(s3Client, secretProvider);

    const bucket = imageRequest.parseImageBucket(event, RequestTypes.THUMBOR);
    // Assert
    expect(bucket).toEqual("test-bucket")
  })
```

```
it("should parse bucket-name from first part in thumbor request and return it when chaining multiple filters", () => {
    // Arrange
    const event = { path: "/filters:grayscale():rotate(180)/s3:test-bucket/test-image-001.jpg" };
    process.env.SOURCE_BUCKETS = "allowedBucket001, test-bucket";

    // Act
    const imageRequest = new ImageRequest(s3Client, secretProvider);

    const bucket = imageRequest.parseImageBucket(event, RequestTypes.THUMBOR);
    // Assert
    expect(bucket).toEqual("test-bucket")
  })
```

Reply:
IMO this should be covered by a totally different test-suite since it's not really part of that function

Reply:
ok, thats fair enough